src/backend/vulkan/src/window.rs

```diff
@@ -244,10 +243,10 @@ impl Instance {
         let surface = {
             let mac_os_loader = mvk::MacOSSurface::new(entry, &self.raw.0);
             let info = vk::MacOSSurfaceCreateInfoMVK {
-                s_type: vk::StructureType::WIN32_SURFACE_CREATE_INFO_KHR,
+                s_type: vk::StructureType::MACOS_SURFACE_CREATE_INFO_M,
                 p_next: ptr::null(),
                 flags: vk::MacOSSurfaceCreateFlagsMVK::empty(),
-                p_view: view,
+                p_view: layer as *mut c_void,
```

Comment:
I don't think this is correct. Here is what the extension [documentation](https://www.khronos.org/registry/vulkan/specs/1.1-extensions/man/html/VkMacOSSurfaceCreateInfoMVK.html) says:
> pView is a reference to a NSView object which will display this surface. This NSView must be backed by a CALayer instance of type CAMetalLayer.

Reply:
Good catch! This is fixed now. MoltenVK didn't seem to mind.

Reply:
it doesn't bother to specify its own extension properly :/